src/solver/modelConverter/modelConverter.cpp

```diff
@@ -0,0 +1,211 @@
+          convertType(variable.variable_type)});
+    }
+    return variables;
+}
+
+/**
+ * \brief Converts constraints from ModelParser::Model to ObjectModel::Constraint.
+ *
+ * \param model The ModelParser::Model object containing constraints.
+ * \return A vector of ObjectModel::Constraint objects.
+ */
+std::vector<Antares::Solver::ObjectModel::Port> convertPorts(
+  const Antares::Solver::ModelParser::Model& model)
+{
+    std::vector<Antares::Solver::ObjectModel::Port> ports;
+    for (const auto& port: model.ports)
+    {
+        // ports.emplace_back(Antares::Solver::ObjectModel::Port{port.name, port.type});
+    }
+    return ports;
+}
+
+std::vector<Antares::Solver::ObjectModel::Constraint> convertConstraints(
+  const Antares::Solver::ModelParser::Model& model)
+{
+    std::vector<Antares::Solver::ObjectModel::Constraint> constraints;
+    for (const auto& constraint: model.constraints)
+    {
+        constraints.emplace_back(Antares::Solver::ObjectModel::Constraint{
+          constraint.id,
+          Antares::Solver::ObjectModel::Expression{constraint.expression}});
+    }
+    return constraints;
+}
+
+/**
+ * \brief Converts models from ModelParser::Library to ObjectModel::Model.
+ *
+ * \param library The ModelParser::Library object containing models.
+ * \return A vector of ObjectModel::Model objects.
+ */
+std::vector<Antares::Solver::ObjectModel::Model> convertModels(
+  const Antares::Solver::ModelParser::Library& library)
+{
+    std::vector<Antares::Solver::ObjectModel::Model> models;
+    for (const auto& model: library.models)
+    {
+        Antares::Solver::ObjectModel::ModelBuilder modelBuilder;
+        std::vector<Antares::Solver::ObjectModel::Parameter> parameters = convertParameters(model);
+        std::vector<Antares::Solver::ObjectModel::Variable> variables = convertVariables(model);
+        std::vector<Antares::Solver::ObjectModel::Port> ports = convertPorts(model);
+        std::vector<Antares::Solver::ObjectModel::Constraint> constraints = convertConstraints(
+          model);
+
+        auto modelObj = modelBuilder.withId(model.id)
+                          .withObjective(Antares::Solver::ObjectModel::Expression{model.objective})
+                          .withParameters(std::move(parameters))
+                          .withVariables(std::move(variables))
+                          .withPorts(std::move(ports))
+                          .withConstraints(std::move(constraints))
+                          .build();
+        models.emplace_back(std::move(modelObj));
+    }
+    return models;
+}
+
+/**
+ * \brief Converts a ModelParser::Library object to an ObjectModel::Library object.
+ *
+ * \param library The ModelParser::Library object to convert.
+ * \return The corresponding ObjectModel::Library object.
+ */
+Antares::Solver::ObjectModel::Library convert(const Antares::Solver::ModelParser::Library& library)
+{
+    Antares::Solver::ObjectModel::LibraryBuilder builder;
```

Comment:
At first glance, I wonder why we would need a **builder** to build a **ObjectModel :: Library**, which is external to the class **ObjectModel :: Library** : I mean, constructor are made for this, right ? So why not use them ? 
But maybe there is a strong reason for using an external builder.

This comment apply to other builder classes in this file

Reply:
https://stackoverflow.com/questions/29881135/difference-between-builder-pattern-and-constructor

Since convert() is the only client it doesn't matter much but:

Mainly because a constructor to those objects could become hard to manage in the long time.
Model has 6 members, you can define default value with empty map or string for each of them. However if a client only want to provide one of the later parameters in the list, they need to still provide empty values for the first ones. Only values after the one provided by the client are defaulted.

If a attribute need to be added of removed, the builder implementation can change and no change is made to the object constructor or API.

In the same way it encapsulate logics in different with method. Those method can be overloaded, deleted, or added without impacting the object API and constructor.

Reply:
ok, seems compelling